src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -696,5 +599,32 @@ private static bool TryGetFormattingSpan(int absoluteIndex, IReadOnlyList<Format
 
             return false;
         }
+
+        private static IReadOnlyDictionary<string, IReadOnlyList<RazorFormatOnTypeProvider>> BuildFormatOnTypeProviderMappings(IEnumerable<RazorFormatOnTypeProvider> formatOnTypeProviders)
```

Comment:
Would you be able to get rid of this entire method if you add a `CanHandle` method in `RazorFormatOnTypeProvider`? That pattern would look cleaner in my opinion.

Reply:
Kind of. We still need to broadcast a static list of trigger characters that are the source of truth for if a format on type provider can handle a character. For instance, in the "CanHandle" method does it just become a `if (TriggerCharacter == XYZ)`? If that's the case we could get rid of the method and add that logic in here. That being said iterating a flat list of providers and performing that check would probably be cleaner than this 😄

Reply:
> For instance, in the "CanHandle" method does it just become a if (TriggerCharacter == XYZ)?

For now yes. But I could see how the logic for some of it could be more than that. For example, we expose an option to enable/disable a specifc part of formatting etc. 

As for the static list of trigger characters, it can just be computed in the constructor by iterating the provider list.

Reply:
> For now yes. But I could see how the logic for some of it could be more than that. For example, we expose an option to enable/disable a specifc part of formatting etc.

Why would that be better than just doing that check in `TryFormatOnType` though?

Reply:
Only if it gets rid of extra logic in `RazorFormattingService` layer. If it looks like it doesn't change anything but just moves logic around then `TryFormatOnType` is perfectly fine.

Reply:
Lets see how you feel about the next iteration. I'm always on board for simplification 😄